File: M3 Intermediate DSA/src/L9_Arrays_Sliding_Window/Assignment2.java
package L9_Arrays_Sliding_Window;

public class Assignment2 {

  public int solve(int[] A, int B) {
    int N = A.length;
    int K = 0;

    for (int el: A) {
      if (el <= B) {
        K++;
      }
    }

    if (K == 0 || K == 1 || K == N) {
      return 0;
    }

    int badElements = 0;

    for (int index = 0; index < K; index++) {
      if (A[index] > B) {
        badElements++;
      }
    }

    int s = 1;
    int e = K;
    int ans = badElements;

    while (e < N) {
      if (A[s-1] > B) {
        badElements--;
      }
      if (A[e] > B) {
        badElements++;
      }

      ans = Math.min(ans, badElements);
      e++;
      s++;
    }

    return ans;
  }

}

File: M3_Intermediate_DSA/L7_Sliding_Window/assignment2.go
package main

func SolveA2(A []int, B int) int {
	k := 0
	N := len(A)

	for index := range A {
		if A[index] <= B {
			k++
		}
	}

	if k == 0 || k == 1 || k == N {
		return 0
	}

	badElement := 0

	for index := 0; index < k; index++ {
		if A[index] > B {
			badElement++
		}
	}

	s := 1
	e := k
	ans := badElement

	for e < N {
		if A[s-1] > B {
			badElement--
		}
		if A[e] > B {
			badElement++
		}
		s++
		e++

		if badElement < ans {
			ans = badElement
		}
	}

	return ans
}
